src/value/borrowed.rs

```diff
@@ -97,19 +98,41 @@ impl<'val> Sequence for BorrowedSequence<'val> {
     fn iter<'a>(&'a self) -> Box<dyn Iterator<Item = &'a Self::Element> + 'a> {
         Box::new(self.children.iter())
     }
+
+    fn get(&self, index: usize) -> Option<&Self::Element> {
+        if index > self.children.len() {
+            None
+        } else {
+            Some(&self.children[index])
+        }
+    }
```

Comment:
I'm late to the party here, but this method body can be replaced with `self.children.get(index)`, which already returns an `Option` wrapping the value at that index.